pkg/loop/internal/relayer/pluginprovider/contractreader/contract_reader.go

```diff
@@ -96,6 +100,21 @@ func EncodeVersionedBytes(data any, version EncodingVersion) (*pb.VersionedBytes
 		if err != nil {
 			return nil, fmt.Errorf("%w: %w", types.ErrInvalidType, err)
 		}
+	case ValuesEncodingVersion:
+		switch val := data.(type) {
+		case values.Value:
+			bytes, err = proto.Marshal(values.Proto(val))
+			if err != nil {
+				return nil, fmt.Errorf("%w: %w", types.ErrInvalidType, err)
+			}
+		case *values.Value:
+			bytes, err = proto.Marshal(values.Proto(*val))
+			if err != nil {
+				return nil, fmt.Errorf("%w: %w", types.ErrInvalidType, err)
+			}
+		default:
+			return nil, fmt.Errorf("%w: data is not of Value type", types.ErrInvalidType)
```

Comment:
This is wrong, you should wrap it in a values.Value.  The EVM codec ins't going to give the chain reader a values.Value.  Wrapping it can then take care of all cases, you don't need to worry about what it is.

Reply:
so it does, replaced with wrap